components/header/header.tsx

```diff
@@ -1,50 +1,125 @@
 import Link from 'next/link';
+import { useState } from 'react';
 
-import Categories from '../categories/categories';
 import Button, { BUTTON_COLORS } from '../button/button';
 import FiniamLogo from '../finiam-logo/finiam-logo';
 
 import styles from './header.module.scss';
 
-type HeaderProps = {
-  showCategories: boolean;
-};
+export default function Header() {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const toggleMenu = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
 
-export default function Header({ showCategories }: HeaderProps) {
   return (
     <header className={styles.header}>
-      <nav className={styles.navbar}>
-        <Link href='/'>
-          <a className='navbar-logo'>
-            <FiniamLogo />
-          </a>
-        </Link>
-        <div className={styles['navbar-wrapper']}>
-          <div className='navbar-links'>
-            <a href='https://finiam.com/about' className={styles['navbar-link']}>
-              About
-            </a>
-            <a href='https://finiam.com/work' className={styles['navbar-link']}>
-              Work
-            </a>
-            <Link href='/'>
-              <a className={styles['navbar-link']}>Blog</a>
-            </Link>
-            <a href='https://finiam.com/contact' className={styles['navbar-link']}>
-              Contact
+      <div className={`${styles['header-wrapper']} ${isMenuOpen ? styles.open : ''}`}>
+        <nav className={styles['nav-main']}>
+          <Link href='/'>
+            <a className={`${styles['nav-logo']} ${isMenuOpen ? styles.light : styles.dark}`}>
+              <FiniamLogo />
             </a>
+          </Link>
+
+          <div onClick={toggleMenu} className={styles['menu-toggle']}>
```

Comment:
As this is an interactive element that you can click, it should be a `<button type="button>`. That also improves the accessibility, so you can tab to it and press Enter, for instance (not an issue for mobile, but still valid)